Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/questions-dashboard/questions-dashboard.component.ts

```diff
@@ -105,32 +140,63 @@ export class QuestionsDashboardComponent implements OnInit {
      */
     categoryWiseFilter(category: Category) {
         this.loader = true;
-        this.selectedCategory = category;
-        this.countQuestion();
-        this.questionDisplay = this.filterQuestion(this.question, this.selectedCategory, this.selectedDifficulty, this.matchString);
-        this.loader = false;
+        this.difficultyLevel = 'All';
+        this.selectedDifficulty = DifficultyLevel[this.difficultyLevel];
         window.scrollTo(0, 0);
+        this.categroyId = category.id;
+        this.countTheQuestion();
+        this.index = 0;
+        this.isAllQuestionsHaveCome = false;
+        this.questionsService.getQuestions(this.index, this.categroyId, this.difficultyLevel, this.matchString).subscribe((questionsList) => {
+            this.questionDisplay = questionsList;
+            this.loader = false;
+            this.matchString = '';
+            this.index++;
+        });
+        this.selectedCategory = category;
     }
 
     /**
      * To select the DifficultyLevel
      * @param difficulty
      */
     difficultyWiseSearch(difficulty: string) {
+        this.loader = true;
         this.selectedDifficulty = DifficultyLevel[difficulty];
-        this.questionDisplay = this.filterQuestion(this.question, this.selectedCategory, this.selectedDifficulty, this.matchString);
+        window.scrollTo(0, 0);
+        this.index = 0;
+        this.difficultyLevel = difficulty;
+        this.isAllQuestionsHaveCome = false;
+        this.questionsService.getQuestions(this.index, this.categroyId, this.difficultyLevel, this.matchString).subscribe((questionsList) => {
+            this.questionDisplay = questionsList;
+            this.index++;
+            this.loader = false;
+        });
     }
 
     /**
      * To get the Search criteria from the user
      * @param matchString
      */
     getQuestionsMatchingSearchCriteria(matchString: string) {
-        if (matchString.length > 2 || matchString.length === 0) {
-            this.matchString = matchString;
-            this.countQuestion();
-            this.questionDisplay = this.filterQuestion(this.question, this.selectedCategory, this.selectedDifficulty, this.matchString);
+        this.matchString = matchString;
+        if (matchString.trim().length > 2) {
+            this.index = 0;
+            this.isAllQuestionsHaveCome = false;
+            this.questionsService.getQuestions(this.index, this.categroyId, this.difficultyLevel, this.matchString).subscribe((questionsList) => {
+                this.questionDisplay = questionsList;
+                this.countTheQuestion();
```

Comment:
There is a difference calling it in if and else both, in if matchString has some value to search which is passed to the server and in else its matchString value is "" which is again passed to server